pkg/storage/secret/metadata/keeper_store.go

```diff
@@ -404,7 +411,7 @@ func (s *keeperMetadataStorage) validateSecureValueReferences(ctx context.Contex
 	}()
 
 	thirdPartyKeepers := make([]*keeperDB, 0)
-	for rows.Next() {
+	for keepersRows.Next() {
```

Comment:
The actual problem, we reused the `rows` var, which was exhausted, so it would return false immediately.